### PR TITLE
Feature mes 2031 office basic layout

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES-2031 - initial basic page layout for office page

NOTE: unit tests have not been done as part of this PR - they will be added in a subsequent PR. 

## Pull Request checklist:

- [X] [WIP] tag removed from PR title
- [X] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [ ] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval
fullsizeoutput_e

### CODE REVIEW

File: src/modules/tests/examiner/examiner.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  staffNumber: null,
+};
+
+export function examinerReducer(
+  state = initialState,
+  action: examinerActions.Types,
+): Examiner {
+  switch (action.type) {
+    case examinerActions.POPULATE_EXAMINER:
+      console.log(`payload ${JSON.stringify(action.payload)}`);
```
Comment (johnmorleybjss) [inline src/modules/tests/examiner/examiner.reducer.ts]: Logs to be removed.

File: src/modules/tests/examiner/examiner.selector.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { Examiner } from '@dvsa/mes-test-schema/categories/B';
+
+export const getStaffNumber = (examiner: Examiner) => examiner.staffNumber || '';
```
Comment (johnmorleybjss) [inline src/modules/tests/examiner/examiner.selector.ts]: Assuming the default is just to keep the compiler happy?

File: src/modules/tests/examiner/examiner.actions.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Action } from '@ngrx/store';
+import { Examiner } from '@dvsa/mes-test-schema/categories/B';
+
+export const POPULATE_EXAMINER = '[JournalEffects] Populate Examiner';
```
Comment (johnmorleybjss) [inline src/modules/tests/examiner/examiner.actions.ts]: Prefix (`JournalEffects`) is wrong.

File: src/modules/tests/test-centre/test-centre.actions.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Action } from '@ngrx/store';
+import { TestCentre } from '@dvsa/mes-test-schema/categories/B';
+
+export const POPULATE_TEST_CENTRE = '[JournalEffects] Populate test centre';
```
Comment (johnmorleybjss) [inline src/modules/tests/test-centre/test-centre.actions.ts]: `JournalEffects`

File: src/modules/tests/test-centre/test-centre.reducer.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  costCode: null,
+};
+
+export function testCentreReducer(
+  state = initialState,
+  action: testCentreActions.Types,
+): TestCentre {
+  switch (action.type) {
+    case testCentreActions.POPULATE_TEST_CENTRE:
+      console.log(`payload ${JSON.stringify(action.payload)}`);
```
Comment (johnmorleybjss) [inline src/modules/tests/test-centre/test-centre.reducer.ts]: Log.

File: src/modules/tests/test-slot-attributes/test-slot-attributes.actions.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Action } from '@ngrx/store';
+import { TestSlotAttributes } from '@dvsa/mes-test-schema/categories/B';
+
+export const POPULATE_TEST_SLOT_ATTRIBUTES = '[JournalEffects] Populate Test slot attributes';
```
Comment (johnmorleybjss) [inline src/modules/tests/test-slot-attributes/test-slot-attributes.actions.ts]: `JournalEffects`

File: src/modules/tests/test-slot-attributes/test-slot-attributes.reducer.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  specialNeeds: null,
+};
+
+export function testSlotsAttributesReducer(
+  state = initialState,
+  action: testSlotAttributesActions.Types,
+): TestSlotAttributes {
+  switch (action.type) {
+    case testSlotAttributesActions.POPULATE_TEST_SLOT_ATTRIBUTES:
+      console.log(`payload ${JSON.stringify(action.payload)}`);
```
Comment (johnmorleybjss) [inline src/modules/tests/test-slot-attributes/test-slot-attributes.reducer.ts]: Log.

File: src/modules/tests/examiner/examiner.reducer.ts (hunk 1 above)
Comment (Jason-Pullen) [inline src/modules/tests/examiner/examiner.reducer.ts]: drop console log

File: src/modules/tests/test-summary/test-summary.actions.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+export class RouteNumberChanged implements Action {
+  readonly type = ROUTE_NUMBER_CHANGED;
+  constructor(public routeNumber: number) {}
+}
+
+export class DebriefWitnessed implements Action {
+  readonly type = DEBRIEF_WITNESSED;
+  constructor() {}
```
Comment (johnmorleybjss) [inline src/modules/tests/test-summary/test-summary.actions.ts]: If there's no arguments being declared, the constructor can be omitted.

File: src/modules/tests/test-summary/test-summary.reducer.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
+import { TestSummary } from '@dvsa/mes-test-schema/categories/B';
+import { createFeatureSelector } from '@ngrx/store';
+import * as testSummaryActions from '../../../modules/tests/test-summary/test-summary.actions';
+// import { has } from 'lodash';
```
Comment (johnmorleybjss) [inline src/modules/tests/test-summary/test-summary.reducer.ts]: Comment.

File: src/modules/tests/test-summary/test-summary.selector.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import { TestSummary } from '@dvsa/mes-test-schema/categories/B';
+
+export const getRouteNumber = (testSummary: TestSummary): number => testSummary.routeNumber;
+export const getCandidateDescription = (testSummary: TestSummary): string => testSummary.candidateDescription;
+export const getAdditionalInformation = (testSummary: TestSummary): string => testSummary.additionalInformation;
+export const isIdentificationLicense = (testSummary: TestSummary): boolean => {
```
Comment (johnmorleybjss) [inline src/modules/tests/test-summary/test-summary.selector.ts]: You could simplify this by converting the return in block on the right of the arrow to an expression. Also, no need for `|| false`, the equality expression does the same thing on its own. Same thing for quite a few of these selectors.

File: src/pages/journal/components/test-outcome/test-outcome.ts [hunk 11]
```diff
@@ -41,6 +41,7 @@ export class TestOutcomeComponent {
   }
 
   startTest() {
+    console.log(`starting test ${this.slotId}`);
```
Comment (johnmorleybjss) [inline src/pages/journal/components/test-outcome/test-outcome.ts]: Log.

File: src/pages/journal/journal.effects.ts [hunk 12]
```diff
@@ -218,4 +223,23 @@ export class JournalEffects {
       ];
     }),
   );
+
+  extractTestSlotAttributes = (slotData): TestSlotAttributes => {
+    console.log(` slotdata ${JSON.stringify(slotData)}`);
```
Comment (johnmorleybjss) [inline src/pages/journal/journal.effects.ts]: Log.

File: src/pages/journal/journal.effects.ts [hunk 13]
```diff
@@ -218,4 +223,23 @@ export class JournalEffects {
...
+      slotId: slotData.slotDetail.slotId,
+      start: slotData.slotDetail.start,
+      specialNeeds: slotData.booking.application.specialNeeds,
+      vehicleSlotType: slotData.vehicleSlotType,
+      extendedTest: slotData.booking.application.extendedTest,
+    };
+  }
+
+  extractTestCentre = (slotData): TestCentre => {
+    console.log(`slotdata ${JSON.stringify(slotData)}`);
```
Comment (johnmorleybjss) [inline src/pages/journal/journal.effects.ts]: Log.

File: src/pages/office/office.html [hunk 14]
```diff
@@ -1,52 +1,279 @@
...
-              <ion-col col-34>
-                <h4 class="fault-heading">ETA</h4>
-              </ion-col>
-              <ion-col align-self-start>
-                <div id="etaFaults">{{pageState.etaFaults$ | async}}</div>
+<ion-content>
+        <ion-row class="mes-full-width-card" id="candidate-section">
+            <ion-col col-64><h2 id="candidate-name">{{pageState.candidateName$ | async}}</h2><h3  id="candidate-driver-number">{{pageState.candidateDriverNumber$ | async}}</h3></ion-col>
+            <ion-col></ion-col>
+            <ion-col col-auto style="float:right; text-align:right"><h2>{{pageState.startTime$ | async}}</h2><h3>test outcome:
```
Comment (johnmorleybjss) [inline src/pages/office/office.html]: Inline styles should go.

File: src/pages/office/office.html [hunk 15]
```diff
@@ -1,52 +1,279 @@
...
-                <h4 class="fault-heading">ETA</h4>
-              </ion-col>
-              <ion-col align-self-start>
-                <div id="etaFaults">{{pageState.etaFaults$ | async}}</div>
+<ion-content>
+        <ion-row class="mes-full-width-card" id="candidate-section">
+            <ion-col col-64><h2 id="candidate-name">{{pageState.candidateName$ | async}}</h2><h3  id="candidate-driver-number">{{pageState.candidateDriverNumber$ | async}}</h3></ion-col>
+            <ion-col></ion-col>
+            <ion-col col-auto style="float:right; text-align:right"><h2>{{pageState.startTime$ | async}}</h2><h3>test outcome: 
+              <span [class.mes-green]="pageState.isPassed$ | async" [class.mes-red]="!(pageState.isPassed$ | async)">{{pageState.testOutcome$ | async}}</span></h3>
```
Comment (johnmorleybjss) [inline src/pages/office/office.html]: `ngClass` with a ternary would be more concise here.

File: src/modules/tests/test-slot-attributes/test-slot-attributes.reducer.ts (hunk 7 above)
Comment (Jason-Pullen) [inline src/modules/tests/test-slot-attributes/test-slot-attributes.reducer.ts]: remove

File: src/modules/tests/test-centre/test-centre.reducer.ts (hunk 5 above)
Comment (Jason-Pullen) [inline src/modules/tests/test-centre/test-centre.reducer.ts]: remove

File: src/pages/journal/components/test-outcome/test-outcome.ts (hunk 11 above)
Comment (Jason-Pullen) [inline src/pages/journal/components/test-outcome/test-outcome.ts]: remove

File: src/pages/journal/journal.effects.ts (hunk 12 above)
Comment (Jason-Pullen) [inline src/pages/journal/journal.effects.ts]: remove

File: src/pages/journal/journal.effects.ts (hunk 13 above)
Comment (Jason-Pullen) [inline src/pages/journal/journal.effects.ts]: remove

File: src/pages/office/office.html (hunk 15 above)
Comment (rossfauldsbjss) [reply src/pages/office/office.html]: done

File: src/pages/office/office.html (hunk 14 above)
Comment (rossfauldsbjss) [reply src/pages/office/office.html]: done

File: src/pages/journal/journal.effects.ts (hunk 13 above)
Comment (rossfauldsbjss) [reply src/pages/journal/journal.effects.ts]: removed

File: src/pages/journal/journal.effects.ts (hunk 12 above)
Comment (rossfauldsbjss) [reply src/pages/journal/journal.effects.ts]: removed

File: src/pages/journal/components/test-outcome/test-outcome.ts (hunk 11 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/test-outcome/test-outcome.ts]: removed

File: src/modules/tests/test-centre/test-centre.reducer.ts (hunk 5 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-centre/test-centre.reducer.ts]: removed

File: src/modules/tests/test-centre/test-centre.actions.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-centre/test-centre.actions.ts]: changed

File: src/modules/tests/examiner/examiner.actions.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply src/modules/tests/examiner/examiner.actions.ts]: changed

File: src/modules/tests/examiner/examiner.selector.ts (hunk 2 above)
Comment (rossfauldsbjss) [reply src/modules/tests/examiner/examiner.selector.ts]: believe there was an issue when running in web mode.

File: src/modules/tests/examiner/examiner.reducer.ts (hunk 1 above)
Comment (rossfauldsbjss) [reply src/modules/tests/examiner/examiner.reducer.ts]: removed

**Pull request merged**